api/envoy/service/ratelimit/v3/rls.proto

```diff
@@ -52,8 +52,31 @@ message RateLimitRequest {
   uint32 hits_addend = 3;
 }
 
+// Cacheable quota for responses.
+// Quota can be granted at different levels: either for each individual descriptor or for the whole descriptor set.
+// This is a certain number of requests over a period of time.
+// The client may cache this result and apply the effective RateLimitResponse to future matching
+// requests containing a matching  without querying rate limit service.
+//
+// When quota expires due to timeout, a new RLS request will also be made.
+// The implementation may choose to preemptively query the rate limit server for more quota on or
+// before expiration or before the available quota runs out.
+// [#not-implemented-hide:]
+message Quota {
+  // Number of matching requests granted in quota. Must be 1 or more.
+  uint32 requests = 1 [(validate.rules).uint32 = {gt: 0}];
+
+  oneof expiration_specifier {
+    // Point in time at which the quota expires.
+    google.protobuf.Timestamp valid_until = 2;
+  }
+
+  // The unique id that is associated with each Quota.
```

Comment:
I would add clarification that this can be used to associate the same quota with different descriptors or descriptor sets.

Reply:
Good point! Done